ocs_ci/utility/kms.py

```diff
@@ -0,0 +1,503 @@
+        self.vault_root_token = self.vault_conf["VAULT_ROOT_TOKEN"]
+
+    def update_vault_env_vars(self):
+        """
+        In order to run vault CLI we need following env vars
+        VAULT_ADDR, VAULT_TOKEN
+
+        """
+        os.environ["VAULT_ADDR"] = f"https://{self.vault_server}:{self.port}"
+        os.environ["VAULT_TOKEN"] = self.vault_root_token
+
+    def create_ocs_vault_resources(self):
+        """
+        This function takes care of creating ocp resources for
+        secrets like ca cert, client cert, client key and vault token
+        Assumption is vault section in AUTH file contains base64 encoded
+        (with padding) ca, client certs, client key and vault path token
+
+        """
+        if not config.ENV_DATA.get("VAULT_SKIP_VERIFY"):
+            # create ca cert secret
+            ca_data = templating.load_yaml(constants.EXTERNAL_VAULT_CA_CERT)
+            self.ca_cert_name = config.ENV_DATA.get(
+                "VAULT_CACERT", constants.VAULT_DEFAULT_CA_CERT
+            )
+            ca_data["metadata"]["name"] = self.ca_cert_name
+            ca_data["data"]["cert"] = self.ca_cert_base64
+            self.create_resource(ca_data, prefix="ca")
+
+            # create client cert secret
+            client_cert_data = templating.load_yaml(
+                constants.EXTERNAL_VAULT_CLIENT_CERT
+            )
+            self.client_cert_name = config.ENV_DATA.get(
+                "VAULT_CLIENT_CERT", constants.VAULT_DEFAULT_CLIENT_CERT
+            )
+            client_cert_data["metadata"]["name"] = self.client_cert_name
+            client_cert_data["data"]["cert"] = self.client_cert_base64
+            self.create_resource(client_cert_data, prefix="clientcert")
+
+            # create client key secert
+            client_key_data = templating.load_yaml(constants.EXTERNAL_VAULT_CLIENT_KEY)
+            self.client_key_name = config.ENV_DATA.get(
+                "VAULT_CLIENT_KEY", constants.VAULT_DEFAULT_CLIENT_KEY
+            )
+            self.client_key_name["metadata"]["name"] = self.client_key_name
+            client_key_data["data"]["key"] = self.client_key_base64
+            self.create_resource(client_key_data, prefix="clientkey")
+
+        # create oc resource secret for token
+        token_data = templating.load_yaml(constants.EXTERNAL_VAULT_KMS_TOKEN)
+        # token has to base64 encoded (with padding)
+        token_data["data"]["token"] = base64.b64encode(
+            # encode() because b64encode expects a byte type
+            self.vault_path_token.encode()
+        ).decode()  # decode() because b64encode returns a byte type
+        self.create_resource(token_data, prefix="token")
+
+        # create ocs-kms-connection-details
+        connection_data = templating.load_yaml(
+            constants.EXTERNAL_VAULT_KMS_CONNECTION_DETAILS
+        )
+        connection_data["data"][
+            "VAULT_ADDR"
+        ] = f"https://{self.vault_server}:{self.port}"
```

Comment:
instead of forming again, can we get from env vars. already updated the VAULT_ADDR in https://github.com/red-hat-storage/ocs-ci/pull/3791/files#diff-5c36a978fe9387d4958d42a61c55682c2f9abdbcd4546cb8e25371a0ae017ee9R128

Reply:
@vavuthu Thanks, I missed this. done

Reply:
done